### PR TITLE
fetch entity statement chains

### PR DESCRIPTION
**Pull request by berendsliedrecht**
- **feat: fetch entity statement chain**

### CODE REVIEW

File: packages/core/src/entityStatement/fetchEntityStatementChain.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  entityConfigurations: Array<EntityConfigurationClaims>
+  verifyJwtCallback: VerifyCallback
+}
+
+export const fetchEntityStatementChain = async ({
+  verifyJwtCallback,
+  entityConfigurations,
+}: FetchEntityStatementChainOptions) => {
+  // The trust anchors (i.e. the last item of the list) entity configuration will be used instead of a statement issued by a superior
+  const statements = [entityConfigurations[entityConfigurations.length - 1]]
```
Comment (TimoGlastra) [inline packages/core/src/entityStatement/fetchEntityStatementChain.ts:18]: Do we check/ensure tbis array is non empty?

I think enabling noUncheckedIndexAccess or whatever it is called in TS compilerOptioms might be good

File: packages/core/src/entityStatement/fetchEntityStatementChain.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  const results = await Promise.allSettled(promises)
+  for (const res of results) {
+    if (res.status === 'fulfilled') {
+      statements.push(res.value)
+    } else {
+      throw new OpenIdFederationError(ErrorCode.Validation, res.reason)
+    }
+  }
```
Comment (TimoGlastra) [inline packages/core/src/entityStatement/fetchEntityStatementChain.ts:62]: If it fails when one fails, we can just use Promise.all right?

Would be good to pass an abort signal to the fetch methods so you can cancel all promises instantly when one fails.

**Review by TimoGlastra — APPROVED** [review]

File: packages/core/src/entityStatement/fetchEntityStatementChain.ts (hunk 2 above)
Comment (berendsliedrecht) [reply packages/core/src/entityStatement/fetchEntityStatementChain.ts]: Created a ticket for the abort signal. Optimalization for later I'd say.

File: packages/core/src/entityStatement/fetchEntityStatementChain.ts (hunk 1 above)
Comment (berendsliedrecht) [reply packages/core/src/entityStatement/fetchEntityStatementChain.ts]: Added it which did not help here (but some other places it did) so I just added a check manually that it is non-empty.

**Pull request merged**